awx/main/models/events.py

```diff
@@ -335,64 +346,6 @@ def create_from_data(cls, **kwargs):
     def job_verbosity(self):
         return 0
 
-    def save(self, *args, **kwargs):
-        # If update_fields has been specified, add our field names to it,
-        # if it hasn't been specified, then we're just doing a normal save.
-        update_fields = kwargs.get('update_fields', [])
-        # Update model fields and related objects unless we're only updating
-        # failed/changed flags triggered from a child event.
-        from_parent_update = kwargs.pop('from_parent_update', False)
-        if not from_parent_update:
-            # Update model fields from event data.
-            updated_fields = self._update_from_event_data()
-            for field in updated_fields:
-                if field not in update_fields:
-                    update_fields.append(field)
-
-            # Update host related field from host_name.
-            if hasattr(self, 'job') and not self.host_id and self.host_name:
```

Comment:
Another very large expense is that on *every* event insert, we're doing some type of inventor/host query here to map a host *name* for the current job to a host ID.

This individual queries aren't slow, but this model means that if you `INSERT` 100K events, you're doing 100K `SELECT` queries.

It turns out that this is completely unnecessary - because we compose the inventory ourselves, we just pass the Tower Host ID through hostvars, and include it in the event payload _before we even get to this code_.  That implementation lives here:

https://github.com/ansible/awx/pull/5618/files#diff-9d4ea1dd908b35fb92eaede4bd10bb46R1004